Examples/Tests/MacroExamples/Implementation/ComplexMacros/DictionaryIndirectionMacroTests.swift

```diff
@@ -0,0 +1,108 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import MacroExamplesImplementation
+import SwiftSyntaxMacros
+import SwiftSyntaxMacrosTestSupport
+import XCTest
+
+final class DictionaryStorageMacroTests: XCTestCase {
+  private let macros: [String: Macro.Type] = [
+    "DictionaryStorage": DictionaryStorageMacro.self,
+    "DictionaryStorageProperty": DictionaryStoragePropertyMacro.self,
+  ]
+
+  func testExpansionConvertsStoredProperties() {
+    assertMacroExpansion(
+      """
+      @DictionaryStorage
+      struct Point {
+        var x: Int = 1
+        var y: Int = 2
+      }
+      """,
+      expandedSource: """
+        struct Point {
+          var x: Int = 1 {
+              get {
+                      _storage["x", default: 1] as! Int
+                }
+              set {
+                          _storage["x"] = newValue
+                }
+          }
+          var y: Int = 2 {
+              get {
+                      _storage["y", default: 2] as! Int
+                }
+              set {
+                          _storage["y"] = newValue
+                }
+          }
+
+            var _storage: [String: Any] = [:]
+        }
+        """,
+      macros: macros
+    )
+  }
+
+  func testExpansionWithoutInitializersEmitsError() {
+    assertMacroExpansion(
+      """
+      @DictionaryStorage
+      class Point {
+        let x: Int
+        let y: Int
+      }
+      """,
+      expandedSource: """
+        class Point {
+          let x: Int
+          let y: Int
+
+            var _storage: [String: Any] = [:]
```

Comment:
Since the input to the test files use 2 spaces for indentation, we should pass `indentationWidth: .spaces(2)` to `assertMacroExpansion` so that the expanded source looks nice as well. This applies to all the new test cases.

Reply:
Done, thanks